File: pset6/credit/credit.py
from cs50 import get_string

def main():
    n = get_string("Number: ")
    sum = 0
    val = False
    n = reverse(n)
    
    for i in range(len(n)):
        if i % 2 == 1:
            tmp = int(n[i]) * 2
            if tmp / 10 >= 1:
                k = int(tmp / 10)
                tmp = tmp % 10
                tmp = tmp + k
        else:
            tmp = int(n[i])
        sum = sum + tmp

    
    

    if sum % 10 == 0:
        if n[len(n)-1] == '3' and (n[len(n)-2] == '4' or n[len(n)-2] == '7') and len(n) == 15:
            print("AMEX")
        elif n[len(n)-1] == '5'and len(n) == 16 and (int(n[len(n)-2]) in range(1,6)):
            print("MASTERCARD")
        elif n[len(n)-1] == '4'and (len(n) == 13 or len(n) == 16):
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")
    




def reverse(n): 
  str = "" 
  for i in n: 
    str = i + str
  return str

main()

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n=1;
    int c=1;
    long temp,type1,type2;
    long even = 0;
    long odd = 0;
    long div=10;

    long num = get_long("Number: ");
    while(num/div > 0)
    {
      n++;
      div *= 10;
    }
    //printf("%i\n", n);
    div = 10;
    
    for(int i=0; i<n; i++)
    {
        temp = num%div;
        temp = temp / (div/10);
         if(c==n)
        {
              type1 = temp;
        }
         else if(c==n-1)
        {
              type2 = temp;
              
        }
        if(c%2 == 0)
        {
            temp = temp * 2;
            if(temp/10 != 0)
            {
                long k = temp/10;
                temp = temp % 10;
                temp = temp + k;
            }
            
           
            even = even+temp;
            
        }
        else
        {
            odd = odd+temp;
        }
         
        c++; 
        div = div * 10;
        
        
    }
  
    long val = even + odd;
    //printf("%li\n",val);
    
   if(n==16 && type1 ==5 && (type2==1||type2==2||type2==3||type2==4||type2==5) && val%10 == 0)
   {
       printf("MASTERCARD\n");
   }
   else if(n==15 && type1==3 && (type2==4|| type2==7) && val%10==0)
   {
       printf("AMEX\n");
   }
    else if((n==13 || n==16) && type1==4 && val%10==0)
    {
        printf("VISA\n");
    }
    else
    {
         printf("INVALID\n");
    }
}
